src/main/java/dev/datainmotion/airquality/service/AirQualityService.java

```diff
@@ -1,100 +1,77 @@
 package dev.datainmotion.airquality.service;
 
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
 import dev.datainmotion.airquality.model.Observation;
-import org.apache.pulsar.client.api.SubscriptionType;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Value;
+
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
 import org.springframework.stereotype.Service;
 import org.springframework.web.reactive.function.client.WebClient;
-import reactor.core.publisher.Flux;
-import java.time.Duration;
-import java.util.List;
-import java.util.Random;
-import java.util.StringJoiner;
-import java.util.stream.Collectors;
 
 /**
  * 
  */
 @Service
+@EnableConfigurationProperties(AirQualityProperties.class)
 public class AirQualityService {
-    private static final Logger log = LoggerFactory.getLogger(AirQualityService.class);
 
-    private static final String DEFAULT_ZIPCODE = "08520";
+	private static final Logger log = LoggerFactory.getLogger(AirQualityService.class);
 
-    @Value("${zipcodes}")
-    String[] arrayOfStrings;
+	private final WebClient webClient;
 
-    @Value("${airnowapi.url:http://localhost:8080}")
-    String airnowapi;
+	private final AirQualityProperties apiProperties;
 
-    @Value("${airnowapi.uri}")
```

Comment:
One of the beauties of `@ConfigurationProperties` is ti gets rid of the `@Value` bloat that will ALWAYS happen in app.

Reply:
And another even cooler feature is that it auto-completes in your IDEA when editing `application.yml`